File: working/temas/js/13.js
class Frigo {
    constructor(color = "blanco") {
        this.color = color;
        this.puerta = "cerrada";
        this.temperatura = 20;
    }

    abre() {
        this.puerta = "abierta";
    }

    cierra() {
        this.puerta = "cerrada";
    }

    fijaTemperatura(nuevaTemperatura = -18) {
        this.temperatura = nuevaTemperatura;
    }

    toString() {
        return `Color: ${this.color} | Puerta ${this.puerta} | ${this.temperatura}ºC (${this.paraTemperaturaRecomendada()}ºC)`;
    }

    paraTemperaturaRecomendada() {
        return this.temperatura - 7;
    }

    getTemperatura() {
        return this.temperatura;
    }
}

// Creación de un objeto de la clase Frigo
f = new Frigo("rojo");

// Mostrar la temperatura actual
console.log(f.getTemperatura());

File: working/temas/c++/13.cpp
#include <iostream>
using namespace std;

class Frigo
{
    public:

    string _color;
    string _puerta;
    int _temperatura;

    Frigo(string color="blanco")
    {
        _color = color;
        _puerta = "cerrada";
        _temperatura = 20;
    }

    // Sobrecarga del operador << para la impresión
    friend ostream &operator<<(ostream &os, const Frigo &f)
    {
        os << "Color: " << f._color << " | Puerta: " << f._puerta << " | " << f._temperatura << "ºC";
        return os;
    }

    void abre()
    {
        _puerta = "abierta";
    }

    void cierra()
    {
        _puerta = "cerrada";
    }

    void fija_temperatura(int nueva_temperatura=7)
    {
        _temperatura = nueva_temperatura;
    }

    int para_temperatura_recomendada()
    {
        return _temperatura - 7;
    }

    int temperatura()
    {
        return _temperatura;
    }
};

int main() {
    Frigo mi_frigo("rojo");
    cout << mi_frigo.temperatura() << endl;
}
